docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Extensions` are used by a particular group or several groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be installed and executed from `NEAR CLI`.
+- `Core NEAR CLI` extensions system only allows to introduce top-level commands through its design.
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-sub-account <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - with-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
```

Comment:
So, we will not use the `implicit` word? I like the description, but maybe it should be a part of the command description.

Reply:
This is not about implicit account. When you create a regular sub-account, you usually want to set the full-access key that will be in control of that account, and so here we have two options: ask user to generate the keypair somehow and specify the public key manually, or let us automate that and generate a new keypair. Well, replying here I realized that I was too low-level here, and I want to take some time thinking on how to make these options user-friendly

Reply:
I'm a bit confused as to the difference between `with-plaintext-public-key` and `with-generated-keypair` could someone provide some clarification for me?

Reply:
@BenKurrek 
1. `with-plaintext-public-key` - You are creating an account and adding the existing key by providing its public key as a parameter.
2. `with-generated-keypair` - The key will be generated during the execution of this command and saved locally.

Reply:
ohhh makes sense - this allows you to have multiple accounts using the same key. Thanks for the clarification!